discord/message.py

```diff
@@ -237,6 +237,42 @@ def cached_message(self):
         """Optional[:class:`Message`]: The cached message, if found in the internal message cache."""
         return self._state._get_message(self.message_id)
 
+    async def fetch_message(self):
+        """|coro|
+
+        Retrieves the :class:`Message` from the destination.
+
+        This can only be used by bot accounts.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ------------
+        id: :class:`int`
+            The message ID to look for.
+
+        Raises
+        --------
+        ~discord.NotFound
+            The message was not found.
+        ~discord.ChannelNotFound
```

Comment:
I'm unsure about this. `abc.Messageable.fetch_message` raises `discord.NotFound` due to the 404 so it seems retuning `None` doesn't make sense and this seems like the best exception based on the name, but `BadArgument` doesn't make much sense